### PR TITLE
[FEAT]: Added Char Time Series Chart

### PR DESCRIPTION
**Pull request by AVAniketh0905**
---
title: Issue #None | [FEAT]: Added Char Time Series Chart
---

Discord Username: @dekomori_sanae09

## What type of PR is this? (select all that apply)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 🚧 Breaking Change
- [ ] 🧑‍💻 Code Refactor
- [ ] 📝 Documentation Update

## Description
I have added a chart similar to word toggle history chart ( monkeytype ).

## Key Feature: 
Uses local storage to send data so the same chart can be used for unauthorized users.

## Related Tickets & Documents

- Related Issue #
- Closes #

## QA Instructions, Screenshots, Recordings
Video:

https://github.com/webdevcody/code-racer/assets/95468529/efe9102a-59a9-498a-98a6-1a2c1b7a77af

### UI accessibility concerns?
The graph is unbearable in smaller screens.

## Added/updated tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### CODE REVIEW

File: src/app/race/race.tsx [hunk 1]
```diff
@@ -369,19 +378,20 @@ export default function Race({
       setTotalErrors(totalErrors + 1);
     }
 
-    if (e.key === " " && e.key === code.slice(input.length, input.length + 1)) {
+    if (e.key === code.slice(input.length, input.length + 1)) {
+      const currTime = Date.now();
+      const lastTime = raceTimeStamp[raceTimeStamp.length - 1]?.time || 0;
+      console.log(e.key, input.length, totalErrors, currTime - lastTime);
```
Comment (scape76) [inline src/app/race/race.tsx:416]: you are calculating the cpm from the time between two characters. I believe, it should be done by considering the time between the start of the race and the time when user types in the right character

Comment (scape76) [reply src/app/race/race.tsx:416]: you can also simplify if statement, e.key === code[input.length]

Comment (AVAniketh0905) [reply src/app/race/race.tsx:416]: that cpm it time between charas that was the problem i had its useless to calculate cpm between 2 charas? shuld i just calc avg cpm for that line then?

Comment (scape76) [reply src/app/race/race.tsx:416]: cpm calculating startegy doesnt change, you take the start date, you take the date.now, calculate taken time and based on the input length calculate cpm

**Comment by AVAniketh0905** [comment]
oh nyo!

**Pull request merged**